### ISSUE TITLE
The issue of right padding in DataCollatorForSupervisedDataset.

### ISSUE THREAD

**Issue by Apollo-Yi** [apple/ml-fastvlm#68]
In your finetuning example code(./llava/train/train_qwen.py), I notice that you use right padding for input_ids of different lengths：

class DataCollatorForSupervisedDataset(object):
    """Collate examples for supervised fine-tuning."""

    tokenizer: transformers.PreTrainedTokenizer

    def __call__(self, instances: Sequence[Dict]) -> Dict[str, torch.Tensor]:
        input_ids, labels = tuple([instance[key] for instance in instances]
                                  for key in ("input_ids", "labels"))
        input_ids = torch.nn.utils.rnn.pad_sequence(
            input_ids,
            batch_first=True,
            padding_value=self.tokenizer.pad_token_id)
        labels = torch.nn.utils.rnn.pad_sequence(labels,
                                                 batch_first=True,
                                                 padding_value=IGNORE_INDEX)
        input_ids = input_ids[:, :self.tokenizer.model_max_length]
        labels = labels[:, :self.tokenizer.model_max_length]
        batch = dict(
            input_ids=input_ids,
            labels=labels,
            attention_mask=input_ids.ne(self.tokenizer.pad_token_id),
        )

In common llm training, left padding is usually adopted to avoid issues caused by right padding during autoregressive generation in batch training or inference. Why use left padding here?

**Comment by Apollo-Yi** [comment]
Ok, it's just a interim padding code, we could choose to use left padding or right padding in "prepare_inputs_labels_for_multimodal" method.
